### DISCUSSION TITLE
Required assistance

### DISCUSSION THREAD

**Discussion by SD-4am-ind** [coders-school/cpp-fundamentals#1390] [General] [answered]
I am absolutely beginner in coding and just started coding in cpp language. 
While solving question of finding Prime number using loop: for(int i=2;i<i/2;i++) in this i am unable to understand why we used i/2 instead of n. Someone please help.

**Comment by Shriharsh-Deshmukh** [comment, marked as answer]
Hi @SD-4am-ind,
To find out whether the given number is Prime or Not you can use  for(int i=2;i<=n;i++).
But, it takes more time as it iterate till n th iteration. To minimize the iteration count we use i/2 instead of n as it reduces the iteration count to half. also to reduce beyond that you can use for(int i=2; i*i<=n;i++).

Happy coding 👍🏻

**Reply by SD-4am-ind** [reply]
I see. Thanks once again!
